internal/pkg/agent/cmd/enroll_cmd.go

```diff
@@ -445,24 +453,35 @@ func (c *enrollCmd) prepareFleetTLS() error {
 
 func (c *enrollCmd) daemonReloadWithBackoff(ctx context.Context) error {
 	err := c.daemonReload(ctx)
+	if err != nil &&
```

Comment:
Maybe we can add a debug log about the errors we receive and retry while trying to restart the agent ?